server/oauth2.go

```diff
@@ -408,6 +409,24 @@ func (s *Server) parseAuthorizationRequest(r *http.Request) (*storage.AuthReques
 		return nil, &authErr{"", "", errServerError, ""}
 	}
 
+	// TODO: we don't currently handle prompt but should verify prompt=none
+	if tokenHint != "" {
+		keystore := &storageKeySet{s.storage}
+		payload, err := keystore.VerifySignature(context.TODO(), tokenHint)
+		if err != nil {
+			return nil, &authErr{"", "", errInvalidRequest, "Could not verify id_token_hint"}
```

Comment:
not sure that we actually care if we can verify the signature here... if the token's in a client session from days ago, dex might not not have the keys to verify it anymore. since this is just a 'hint' rather than any proof of authentication, it's probably better to just get the sub claim without caring about the signature. thoughts?